wbooks/src/app/screens/bookDetail/index.js

```diff
@@ -0,0 +1,101 @@
+import React, {Component} from 'react';
+import booksJson from '../../../../resources/books.json';
+import {Link} from 'react-router-dom';
+import './style.css';
+import Comments from './components/Comments';
+import NotFoundPage from '../../components/NotFoundPage';
+
+class BookDetail extends Component {
+
+  constructor({match}){
+    super();
+    this.state = {book: this.getBookInfo(match.params.id),
+      comments: []};
+  }
+
+  getBookInfo(bookId){
+      for(let i= 0; i < booksJson.length; i++) {
+        if(booksJson[i].id == bookId){
+          return booksJson[i];
+        }
+      }
+      return null;
+  }
+
+  getRelatedBooks(book){
+    let relatedBooks = [];
+    for(let i= 0; i < booksJson.length; i++) {  
+        if(booksJson[i].genre === book.genre && booksJson[i].id !== book.id){
+          relatedBooks.push(booksJson[i]);
+        }
+    } 
+    return relatedBooks;
```

Comment:
check the function `array.map`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map